test/e2e.test.ts

```diff
@@ -0,0 +1,69 @@
+/// <reference path="../node_modules/@types/mocha/index.d.ts" />
+
+import * as chai from "chai";
+import * as chaiAsPromised from "chai-as-promised";
+import * as ethers from "ethers";
+import * as dotenv from "dotenv";
+
+import { Config, Deposit, SubConfig } from "../src/types";
+import * as helpers from "../src/helpers";
+import { createInstance } from "../src";
+
+chai.use(chaiAsPromised.default);
+const expect = chai.expect;
+dotenv.config();
+
+describe("Test Custom SDK Logic", () => {
+  const provider = new ethers.providers.StaticJsonRpcProvider(
+    process.env.INFURA_URL,
+    4
+  );
+  it("Runs a full scenario through the SDK", async () => {
+    const config: Config = {
+      wildPoolAddress: "0xE0Bb298Afc5dC12918d02732c824DA44e7D61E2a",
+      lpTokenPoolAddress: "0xe7BEeedAf11eE695C4aE64A01b24F3F7eA294aB6",
+      factoryAddress: "0xb1d051095B6b2f6C93198Cbaa9bb7cB2d607215C",
```

Comment:
just curious, what contract is the factory address referring to?

Reply:
You can check it on [etherscan](https://rinkeby.etherscan.io/address/0xb1d051095B6b2f6C93198Cbaa9bb7cB2d607215C) but basically the factory acts as a manager / coordinator of the pools. When we deploy a new pool it has to be registered with the factory.